.github/workflows/build.yml

```diff
@@ -3,20 +3,77 @@ on:
   push:
     branches:
       - main
+      - lint-gh-action
 
 jobs:
-  install-deps-and-build:
-    concurrency: ci-${{ github.ref }}
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout 🛎️
+    install:
+      runs-on: ubuntu-latest
+      steps:
+      # Checkout the Repo
+      - name: Checkout
         uses: actions/checkout@v3
-      - name: Install, Build and Deploy 🔧
-        run: |
-          yarn
-          yarn build --filter=pie-microsite
-      - name: Deploy
-        uses: peaceiris/actions-gh-pages@v3
+      - name: Setup Node 
+      # Setup Node 16 
+        uses: actions/setup-node@v3
         with:
-          deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
-          publish_dir: ./apps/pie-microsite/dist
+          node-version: 16
+          cache: 'yarn'
+      # Restore node_modules if cache exists. If not, cache is created at end of build.
+      - name: Cache Node Modules
+        id: cache-node-modules
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: node-modules-${{ hashFiles('**/yarn.lock') }}
+      # Run 'yarn' if cache doesn't exist. Use --prefer-offline to download packages from yarn cache folder where possible
+      - name: Install Dependencies
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
+        run: yarn add --cached
```

Comment:
Should this be `yarn install`?

Reply:
I think so - the instructions were a bit weird and specified yarn add. This PR isn't quite ready for review though haha :D